scripts/variantreport_p_sample.rmd

```diff
@@ -150,35 +150,19 @@ deconv <- function (bulk,sig){
   as.vector(rlm_coefficients)
 }
 
-getIdentCols <- function (dataframe) {
-    '* This function is for collapsing the column names of identical columns and saving them as a new column names.
-       I had the difficulty that with all the existing functions I would always would have missed the "parent" of the 
-       following duplicates. This workaround here should be definitly improved with a more R-like version, if possible. *'
-  
-    cols_new <- c()
-    all_dups <- c()
-    for (i in 1:ncol(msig)){
-      dups <- c()
-      dups <- c()
-      parent <- c()
-      for (j in 1:ncol(msig)){
-        if (i!=j && all(msig[,i] == msig[,j])){
-                parent <- colnames(msig[i])
-                if (!(colnames(msig[j]) %in% dups)) {
-                    dups <- c(dups,colnames(msig[j]))
-                }
-        }else{
-          if (!(colnames(msig[i]) %in% cols_new) && (colnames(msig[i]) %in% all_dups)==FALSE) {
-            cols_new <- c(cols_new, colnames(msig[i]))
-          }
-        }
-      }
-      if (is.null(parent)==FALSE && (colnames(msig[i]) %in% dups)==FALSE && (parent %in% all_dups)==FALSE) {
-        cols_new[i] <- paste(parent,paste(dups,collapse =','), sep = ',')
-      }
-      all_dups <- c(all_dups,dups)
+dedupe <- function (df) {
+    transposed <- t (df)
+    dupes <- duplicated (transposed, fromLast=TRUE)
+    allDupes <- dupes | duplicated (transposed, fromLast=FALSE)
+    if (any (allDupes)) {
+        grouped <- colnames (df)[allDupes]
+        groupName <- paste(grouped, collapse = ",")
+        deduped <- df[!dupes]
+        names (deduped)[names (deduped) %in% grouped] <- groupName
+        return (deduped)
+    } else {
+        return (df)
```

Comment:
I'm glad this turned out to be useful!  I bet we could make it slightly more efficient by removing all but one of the collapsed columns before applying `duplicated` for a second time.  This may be an important optimization in case the data frame becomes very large with *many* columns and *many* rows, as we could cut down on the number of column comparisons.

Reply:
Btw: When I used this to develop the dedupMuts I realized that dedupe will not work when we have the case that we have different groups of identical columns. Because dedupe just merges everything which is duplicated in any place. 

I mean when I e.g. maybe would need column names like: "wt - b117 - b1351,p1 - b1651 - b1429,b1427" 
I think with dedupe we would end up in a "wt - b117 - b1351,p1,b1429,b1427 - b1651 - b1351,p1,b1429,b1427" which would be wrong.

Reply:
I'm having a hard time understanding the example column names; what's the separator and how many columns are there in both cases?  And why would it be wrong?  What would the correct column names look like?

Reply:
hmm sorry. :( 

Ok considering the case: >> "wt" - "b117" - "b1351,p1" - "b1651" - "b1429,b1427" <<, 5 columns 
As a product from a DF with previously 7 columns. But "b1351,p1" are identical and "b1429,b1427" are identical. 
But "b1351,p1" IS NOT identical with "b1429,b1427". 

In dedupe you use "duplicated" to identify the identical columns and then you merge all the column names into one single group. That's why I think with this example instead of "b1351,p1" & "b1429,b1427" as two seperate column names, we only will get "b1351,p1,b1429,b1427" as a new column name.

Reply:
But it's just what I observed when I tried to use it as it is for the sigmut dataframe. But maybe I just did smth else different there. 

I want and will investigate this further with some dummy data for the example I described above. But it works right now. This will be a problem later when we have more or different variants. But yes - fixing/double-checking this will be at the top of the list after we have smth deliverable till Friday.